echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/monitor/DockerEventMonitor.java

```diff
@@ -77,7 +80,19 @@ protected boolean isSuccessfulTriggerEvent(final TriggerEvent event) {
   @Override
   protected Function<Trigger, Pipeline> buildTrigger(Pipeline pipeline, TriggerEvent event) {
     DockerEvent dockerEvent = (DockerEvent) event;
-    return trigger -> pipeline.withTrigger(trigger.atTag(dockerEvent.getContent().getTag()));
+    DockerEvent.Content content = dockerEvent.getContent();
+
+    String name = content.getRegistry() + "/" + content.getRepository();
+    String reference = name + ":" + content.getTag();
```

Comment:
Are `registry`, `repository` and `tag` always present on a DockerEvent?

Reply:
Yes, but to confirm this I went looking for where the events are coming from, which lead me to [this code in Igor](https://github.com/spinnaker/igor/blob/f2f0e3c4c6964fcb369af6132b5fc8ef30c147ad/igor-web/src/main/groovy/com/netflix/spinnaker/igor/docker/DockerMonitor.groovy#L134). It already includes artifacts, so we could use that; however, the artifacts are of the type `"docker"`, while the Kubernetes v2 provider is expecting `"docker/image"`.

They need to be the same (unless we change the type here, but that would be terribly ugly and confusing). `"docker"` is slightly older, while `"docker/image"` is more consistent with the naming other artifact types and has been used in the [new documentation](https://www.spinnaker.io/reference/artifacts/in-kubernetes-v2/).

@lwander, @gardleopard, any thoughts on this?

Reply:
As far as I can tell, the artifact in the event is quite different (of type `GenericArtifact` from Igor) from what we expect. Perhaps Igor should be refactored to use kork? Echo already uses `com.netflix.spinnaker.kork.artifacts.model.Artifact` in a number of places. Anyway, that would be a different commit.

I've updated the tests to use `"account"` for the account.